sdk/typescript/runtime/template/package.json

```diff
@@ -6,15 +6,6 @@
     "typescript": "^5.3.2"
   },
   "devDependencies": {
-    "@lifeomic/axios-fetch": "^3.0.1",
-    "adm-zip": "^0.5.10",
-    "env-paths": "^3.0.0",
-    "execa": "^8.0.1",
-    "graphql": "^16.8.0",
-    "graphql-request": "^6.1.0",
-    "graphql-tag": "^2.12.6",
-    "node-color-log": "^10.0.2",
-    "node-fetch": "^3.3.1",
-    "tar": "^6.1.13"
+    "@dagger.io/dagger": "./sdk"
```

Comment:
Why is this a `devDependencies`? From my memory (not a js dev, so sorry):
- `dependencies` is for runtime requirements
- `devDependencies` is for compilation requirements

Reply:
It's more appropriate to be in `dependencies` when considering that _production_ is the runtime container. But for the IDE, you only want `devDependencies`.

Since the runtime container is calling `npm install`, which installs from `dependencies` and `devDependencies`, it works in both cases.

However, locally, ideally, you install via a link so that changes to the module code are immediate. Otherwise, it may require you to call `npm install` on the host whenever you make changes that affect the codegen.

In Python, the runtime container installs manually from `pip install /sdk`, but locally, you should use an editable install with `pip install -e ./sdk` for those reasons.

Reply:
> However, locally, ideally, you install via a link so that changes to the module code are immediate. Otherwise, it may require you to call npm install on the host whenever you make changes that affect the codegen.

You cannot change `sdk` locally since it's dynamically injected by the runtime so any local change in that folder would be undo (except if you update dagger engine like with `./hack/dev`)

Reply:
> It's more appropriate to be in dependencies when considering that production is the runtime container.

It's true and false at the same time, you might have your CI setup in your root project with you other dependencies.
In that case, you don't want to compile your web app or nestJS app with dagger installed in it, since dagger shall only be executed through `dagger` cli so it's already wrapped into the right environment etc

Reply:
> You cannot change `sdk` locally since it's dynamically injected by the runtime so any local change in that folder would be undo (except if you update dagger engine like with `./hack/dev`)

That's not what I meant. If you add a dependency (module), you'll need to rerun codegen. If you don't link, you'll have to `npm install` again locally after `dagger mod use`. That's why I said "whenever you make changes that affect the codegen". It's something that changes `./sdk`.

Reply:
> That's not what I meant. If you add a dependency (module), you'll need to rerun codegen. If you don't link, you'll have to npm install again locally after dagger mod use. That's why I said "whenever you make changes that affect the codegen". It's something that changes ./sdk.

Actually not, the `tsconfig.json` is the one in charge of the inferance, you shouldn't need to npm install.
I'll record an exemple to verify my assumption

Reply:
https://github.com/dagger/dagger/assets/61683879/777e8d0c-59aa-4df3-a1d8-c63f17178000

@helderco See that I'm installing a module, I do not `npm install` but still the auto completion works and type safety as well (even found I forgot the `Promise`) :D

Reply:
Good, thanks for the confirmation, it's good to know.